src/extension/background-script/actions/ln/request.ts

```diff
@@ -0,0 +1,82 @@
+import utils from "~/common/lib/utils";
+import { MessageGenericRequest } from "~/types";
+
+import db from "../../db";
+import state from "../../state";
+
+const request = async (
+  message: MessageGenericRequest
+): Promise<{ data: unknown } | { error: string }> => {
+  const connector = await state.getState().getConnector();
+
+  const { origin, args } = message;
+
+  try {
+    if (!connector.requestMethod) {
+      throw new Error(
+        `${message.args.method} is not supported by your account`
+      );
+    }
+
+    const allowance = await db.allowances
+      .where("host")
+      .equalsIgnoreCase(origin.host)
+      .first();
+
+    if (!allowance?.id) {
+      return { error: "Could not find an allowance for this host" };
+    }
+
+    const permission = await db.permissions
+      .where("host")
+      .equalsIgnoreCase(origin.host)
+      .and((p) => p.method === args.method)
+      .first();
+
+    if (permission && permission.enabled) {
```

Comment:
Yesy, right now there is either a permission saved in the DB or there is none.
Which makes the booleans "enabled"  and "blocked"  obsolete at the moment.

As @bumi said, this was to think ahead because removing entries from the DB is easier than removing when migrating.

When this follow-up ticket is addressed: https://github.com/getAlby/lightning-browser-extension/issues/1726
=> it will take another concepting again what the user can do with the permission (change "enabled" to false? Do something with "blocked"? Or only delete a permission?)

Reply:
Thanks for clarifying. 👍